dozer-cli/src/pipeline/connector_source.rs

```diff
@@ -85,6 +89,7 @@ impl ConnectorSourceFactory {
         connection: Connection,
         runtime: Arc<Runtime>,
         progress: Option<MultiProgress>,
+        shutdown: ShutdownReceiver,
```

Comment:
I thought about that, but that would involve adding a tokio dependency in dozer-core or extracting Source into dozer-cli

Reply:
You're right. We don't want either of that. In that case, we still have two options to prevent `ShutdownReceiver` from getting into the signature of `SimpleOrchestrator::build`.

1. Make `ConnectorSource` respect the "channel close event".
2. Change `shutdown` to a `Option<ShutdownReceiver>` and pass `None` when we don't mean to run the `Dag`.

I understand that you don't like the implicitly of the first option. The second option will complicate the code in `ConnectorSource` even more so I don't like it either. The decision is up to you.

Reply:
2 also makes it possible to misuse the API and get into a state where you are again stuck waiting on the thread.

I think in this case we have to live with the inconvenient API of this PR. I think this part of dozer can do with quite a big refactor and then we can solve this problem as well.

That's an unsatisfactory outcome (I am not satisfied with it either), but so be it.